tests/drivers/build_all/ethernet/prj.conf

```diff
@@ -7,3 +7,8 @@ CONFIG_TEST_RANDOM_GENERATOR=y
 CONFIG_TEST_USERSPACE=y
 
 CONFIG_GPIO=y
+
+CONFIG_SPI_INIT_PRIORITY=50
+CONFIG_ETH_INIT_PRIORITY=50
+CONFIG_MDIO_INIT_PRIORITY=50
+CONFIG_PHY_INIT_PRIORITY=50
```

Comment:
Considering that default priorities are:
* `CONFIG_SPI_INIT_PRIORITY=70`
* `CONFIG_ETH_INIT_PRIORITY=70`
* `CONFIG_PHY_INIT_PRIORITY=70`

It should be enough to use `CONFIG_MDIO_INIT_PRIORITY=70` to make it work for ADIN. I will be able to check it tomorrow.